include/mqtt/property.hpp

```diff
@@ -100,18 +100,9 @@ struct n_bytes_property {
 };
 
 struct binary_property {
-    template <
-        typename Buffer,
-        typename = std::enable_if_t<
-            !std::is_same<
-                std::decay_t<Buffer>,
-                binary_property
-            >::value
-        >
-    >
-    binary_property(property::id id, Buffer&& buf)
+    binary_property(property::id id, buffer buf)
```

Comment:
Did this change help at all with the compile times?

Reply:
No. compile times is not affected by this change. This change is to avoid implicit conversion. Users need to pass `mqtt::buffer` or "abc"_mb explicitly.

See https://github.com/redboltz/mqtt_cpp/pull/337#discussion_r313724629

Am I understand correctly??

Reply:
I try to avoid using universal / forwarding references when the thing being "forwarded" will be stored into a member variable.

I like pushing the responsibility of providing a fully constructed object onto the caller of the function.  It's much less typing, less template instantiation, and causes much less surprises than using forwarding references.

Reply:
Ok. I understand that you think this change is valuable even if compile times is not changed.

Reply:
Right.

I expected a small improvement to compile times, but mostly I thought it was an improvement to the API regardless of whether compile times improved.

This video explains a bit: https://www.youtube.com/watch?v=PNRju6_yn3o

Note that using the universal / forwarding references is *technically* better in terms of (completely unoptimized) runtime performance because it has slightly less calls to the move constructor, but that's not relevant for header-only code. The compiler will *almost certainly* inline the call to the function, and therefore we should already have very efficient outcomes.

Reply:
Implemented.